WcaOnRails/app/views/competitions/_my_competitions_table.html.erb

```diff
@@ -0,0 +1,57 @@
+<% if competitions.length < 1 %>
+  <div class="alert alert-info" role="alert">
+    <%if not_past %>
+      You currently have no upcoming competitions! Check out the <%= link_to 'competitions list', competitions_path %>
+    <% else %>
+      You have no past competitions.
+    <% end %>
+  </div>
+<% else %>
+  <table class="table competitions-table floatThead <%= !not_past ? "table-striped" : "" %>">
+    <thead>
+      <tr>
+        <th>Name</th>
+        <th>Location</th>
+        <th>Date</th>
+        <th></th>
+        <th></th>
+        <th class="big-column"></th>
+      </tr>
+    </thead>
+    <% competitions.each do |competition| %>
+      <% if competition.isConfirmed %>
+        <% if competition.showAtAll %>
+          <% message = "This competition is confirmed and visible." %>
+        <% else %>
+          <% message = "This competition is confirmed but not visible." %>
+        <% end %>
+      <% else %>
+        <% if competition.showAtAll %>
+          <% message = "This competition is not confirmed but is visible." %>
+        <% else %>
+          <% message = "This competition is not confirmed and not visible." %>
+        <% end %>
+      <% end %>
+      <tbody>
+        <tr class="<%= competition.isConfirmed ? "confirmed" : "not-confirmed" %> <%= competition.showAtAll ? "visible" : "not-visible" %> <%= not_past ? "not-past" : "past" %>" data-toggle="tooltip" data-placement="top" data-container="body" title="<%= not_past ? message : "" %>">
```

Comment:
Well, looking at not_past is easier because we don't have to retrieve competition.is_over? every time...